pynicotine/gtkgui/widgets/iconnotebook.py

```diff
@@ -558,12 +552,22 @@ def request_tab_changed(self, page, is_important=False, is_quiet=False):
 
     def remove_tab_changed(self, page):
 
-        tab_label = self.get_tab_label(page)
-        tab_label.remove_changed()
-
         if self.parent_page is not None:
+            if self.window.current_page_id != self.parent_page.id:
+                # Another tab has since been selected within 250ms
+                return
+
+            GLib.source_remove(self.switch_page_delay_timer)
+            self.switch_page_delay_timer = None
             self._remove_unread_page(page)
 
+        if self.get_current_page() != page:
+            # Another page has since been selected within 250ms
+            return
+
+        tab_label = self.get_tab_label(page)
+        tab_label.remove_changed()
```

Comment:
Since we track the timer now, you should be able to undo all these changes, and just set `self.switch_page_delay_timer = None` in remove_tab_changed().